addons/stock/models/stock_package_level.py

```diff
@@ -131,8 +131,7 @@ def _generate_moves(self):
     def create(self, vals):
         result = super(StockPackageLevel, self).create(vals)
         if vals.get('location_dest_id'):
-            result.mapped('move_line_ids').write({'location_dest_id': vals['location_dest_id']})
-            result.mapped('move_ids').write({'location_dest_id': vals['location_dest_id']})
+            result.location_dest_id = result.location_dest_id.get_putaway_strategy(result.package_id.quant_ids.mapped('product_id')).id or result.location_dest_id.id
```

Comment:
:scream: This looks quite wrong to fix the issue we reported, as it computes a new putaway location for the `stock.package_level` instead of updating the `stock.package_level` with the actual result of the putaway computation from the `stock.move.line`.

From my POV, this is more or less how the issue must be solved: https://github.com/grindtildeath/odoo/commit/63a6e0cbbe1a3c1a55cfc6fa1170dc3f717d4a86

Reply:
@grindtildeath it would actually be helpful if you could write  failing test case